library/pkparse.c

```diff
@@ -1343,7 +1361,9 @@ int mbedtls_pk_parse_key( mbedtls_pk_context *pk,
      * also ok and in line with the mbedtls_pk_free() calls
      * on failed PEM parsing attempts. */
 
-    return( MBEDTLS_ERR_PK_KEY_INVALID_FORMAT );
+    if( ret & MBEDTLS_ERR_PLATFORM_FEATURE_UNSUPPORTED )
+        return( MBEDTLS_ERR_PLATFORM_FEATURE_UNSUPPORTED );
```

Comment:
Why does `MBEDTLS_ERR_PLATFORM_FEATURE_UNSUPPORTED` get this special treatment of masking out the high-level error? And similarly in other places why does it get the special treatment of not masking in a high-level error?